apps/cowswap-frontend/src/modules/onchainTransactions/updaters/FinalizeTxUpdater/hooks/useShouldCheckPendingTx.ts

```diff
@@ -12,30 +12,37 @@ export function useShouldCheckPendingTx() {
   const accountLowerCase = account?.toLowerCase() || ''
 
   return useCallback(
-    (tx: EnhancedTransactionDetails) => {
-      if (!lastBlockNumber) return false
-
-      return (
-        tx.from.toLowerCase() === accountLowerCase && !tx.replacementType && shouldCheckPendingTx(lastBlockNumber, tx)
-      )
-    },
+    (tx: EnhancedTransactionDetails) => shouldCheckPendingTx(lastBlockNumber, accountLowerCase, tx),
     [accountLowerCase, lastBlockNumber]
   )
 }
 
-function shouldCheckPendingTx(lastBlockNumber: number, tx: EnhancedTransactionDetails): boolean {
-  if (tx.receipt || !tx.lastCheckedBlockNumber) return false
+function shouldCheckPendingTx(
+  lastBlockNumber: number | undefined,
+  accountLowerCase: string,
+  tx: EnhancedTransactionDetails
+): boolean {
+  const isCurrentAccount = tx.from.toLowerCase() === accountLowerCase
+  const isReplaced = !!tx.replacementType
+  const isTxMined = !!tx.receipt
+
+  if (!isCurrentAccount || isReplaced || isTxMined) return false
+
+  // If a tx was never checked, we should check it
+  if (!tx.lastCheckedBlockNumber) return true
```

Comment:
Improved logic here.
There might be a case, when we have never set `lastCheckedBlockNumber` to a transaction and in the same time we have never got `receipt` for it.
In that case the tx will get stuck into `pending` state.

This fix works only with the change:
https://github.com/cowprotocol/cowswap/pull/4140/files#r1547786433